clusterversion: remove old guidance to delete gates older than "5 releases" ago

### What
Remove the check that [forces the deletion of old version gates as part of creating gates for the upcoming release](https://github.com/cockroachdb/cockroach/blob/master/pkg/clusterversion/keyed_versions.go#L82-L93), i.e. "found versions that are always active because 2.23 is already released; these should be removed".

### Why
- We've intentionally decoupled the two tasks of deleting old gates from minting the upcoming release, so that the release team can focus on minting the release, and delegate the deletion of old gates to those team owners (see process from 22.2 https://github.com/cockroachdb/cockroach/issues/95530)
- The error message is giving incorrect guidance. The `0.0-` gates noted in the `TestVersionsAreValid` failure message are [VPrimordial gates](https://github.com/cockroachdb/cockroach/blob/master/pkg/clusterversion/cockroach_versions.go#L544-L545), which [should never be removed](https://github.com/cockroachdb/cockroach/blob/master/pkg/clusterversion/cockroach_versions.go#L176-L178).
- The old rule assumed that versions more than 5 releases ago had to be removed. But this rule no longer applies, now that we've since [modified the schedule for when we advance minBinary](https://cockroachlabs.atlassian.net/wiki/spaces/RE/pages/3197665321/Schedule+for+advancing+BinaryMinSupportedVersionKey+a.k.a.+minBinary).

### More Context

This [check was triggered](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/12045194?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildDeploymentsSection=true&expandBuildTestsSection=true) while attempting to [add the version gate to mint V23_2](https://github.com/cockroachdb/cockroach/pull/111729/files#diff-8f32466c40e6960b988c87d7e2453c01a591d420af123f0963959da8e580f2e4).

Error message:

```
Failed
=== RUN   TestVersionsAreValid
    cockroach_versions_test.go:26:
          Error Trace:  github.com/cockroachdb/cockroach/pkg/clusterversion/pkg/clusterversion/cockroach_versions_test.go:26
          Error:        Received unexpected error:
                        found versions that are always active because 2.23 is already released; these should be removed:
                        (1) attached stack trace
                          -- stack trace:
                          | github.com/cockroachdb/cockroach/pkg/clusterversion.keyedVersions.Validate
                          |   github.com/cockroachdb/cockroach/pkg/clusterversion/pkg/clusterversion/keyed_versions.go:106
                          | github.com/cockroachdb/cockroach/pkg/clusterversion.TestVersionsAreValid
                          |   github.com/cockroachdb/cockroach/pkg/clusterversion/pkg/clusterversion/cockroach_versions_test.go:26
                          | testing.tRunner
                          |   GOROOT/src/testing/testing.go:1576
                          | runtime.goexit
                          |   src/runtime/asm_amd64.s:1598
                        Wraps: (2) found versions that are always active because 2.23 is already released; these should be removed:
                          | introduced in 22.1: 0.0-2
                          | introduced in 22.1: 0.0-4
                          | introduced in 22.1: 0.0-6
                          | introduced in 22.1: 0.0-8
                          | introduced in 22.1: 0.0-10
                          | introduced in 22.1: 0.0-12
                          | introduced in 22.1: 0.0-424242
                        Error types: (1) *withstack.withStack (2) *errutil.leafError
          Test:         TestVersionsAreValid
--- FAIL: TestVersionsAreValid (0.02s)
```



Jira issue: CRDB-32114

--- a/pkg/clusterversion/keyed_versions.go
+++ b/pkg/clusterversion/keyed_versions.go
@@ -12,8 +12,6 @@ package clusterversion
 
 import (
 	"context"
-	"fmt"
-	"strings"
 
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -77,38 +75,6 @@ func (kv keyedVersions) Validate() error {
 		}
 		// Add to existing entry.
 		byRelease[n-1].vs = append(byRelease[n-1].vs, namedVersion)
-	}
-
-	// Iterate through all versions known to be active. For example, if
-	//
-	//   byRelease = ["19.1", "19.2", "20.1", "20.2"]
-	//
-	// then we know that the current release cycle is 21.1, so mixed version
-	// clusters are running at least 20.2, so anything slotted under 20.1 (like
-	// 20.1-12) and 19.2 is always-on. To avoid interfering with backports, we're
-	// a bit more lenient and allow one more release cycle until validation fails.
-	// In the above example, we would tolerate 20.1-x but not 19.2-x.
-	// Currently we're actually a few versions behind in enforcing a ban on old
-	// versions/upgrades. See #47447.
-	if n := len(byRelease) - 5; n >= 0 {
-		var buf strings.Builder
-		for i, mami := range byRelease[:n+1] {
-			s := "next release"
-			if i+1 < len(byRelease)-1 {
-				nextMM := byRelease[i+1]
-				s = fmt.Sprintf("%d.%d", nextMM.major, nextMM.minor)
-			}
-			for _, nv := range mami.vs {
-				fmt.Fprintf(&buf, "introduced in %s: %s\n", s, nv.Key)
-			}
-		}
-		mostRecentRelease := byRelease[len(byRelease)-1]
-		return errors.Errorf(
-			"found versions that are always active because %d.%d is already "+
-				"released; these should be removed:\n%s",
-			mostRecentRelease.minor, mostRecentRelease.major,
-			buf.String(),
-		)
 	}
 
 	// Check to see that for versions introduced in v21.1 and beyond, the

